app/src/lib/git/status.ts

```diff
@@ -178,13 +197,21 @@ export async function getStatus(
   const mergeHeadFound = await isMergeHeadSet(repository)
 
   // if we have any conflicted files reported by status, let
-  const conflictState = mergeHeadFound
-    ? await getFilesWithConflictMarkers(repository.path)
-    : new Map<string, number>()
+  const conflictDetails = mergeHeadFound
+    ? {
+        conflictCountsByPath: await getFilesWithConflictMarkers(
+          repository.path
+        ),
+        binaryFilepaths: await getBinaryPaths(repository, 'MERGE_HEAD'),
```

Comment:
I'm curious about the use of `MERGE_HEAD` here as the ref. Are you able to share how this helps with the underlying issue?

Reply:
yes! [git's CLI is a little weird around this.](https://git-scm.com/docs/git-diff#git-diff-emgitdiffemltoptionsgtltcommitgt--ltpathgt82308203) supplying `HEAD` (or no `rev` argument) to git diff compares your working tree to `HEAD`. supplying any other `rev` compares that `rev` to `HEAD`. so supplying `MERGE_HEAD` will compare it to `HEAD`.

does that make sense? this line effectively generates the diff between the two `HEAD`s of the merge. (and assumes that a merge is in progress due to the merge head check above.

Reply:
that sounds better to be explicit!